src/main/java/Iris/Iris.java

```diff
@@ -0,0 +1,34 @@
+package Iris;
+
+import java.util.Scanner;
+
+public class Iris {
+    private final Storage taskStorage;
+    private final Parser commandParser;
+    private static ToDoList toDoList;
+
+    public Iris(String filePath) {
+        commandParser = new Parser();
+        taskStorage = new Storage(filePath);
+        toDoList = new ToDoList(taskStorage.loadTask());
+    }
+
+    public void run() {
+        Ui.welcomeMsg();
+        Scanner scanner = new Scanner(System.in);
+        while (scanner.hasNextLine()) {
+            try {
+                String input = scanner.nextLine();
+                commandParser.parseCommand(taskStorage, toDoList, input);
+            } catch (IllegalArgumentException e) {
```

Comment:
You can consider using your IrisException class to handle the exceptions and display the error messages.